improver/blending/utilities.py

```diff
@@ -225,3 +227,99 @@ def _get_cycletime_point(cube: Cube, cycletime: str) -> int64:
         cycletime, time_unit=frt_units, calendar=frt_calendar
     )
     return round_close(cycletime_point, dtype=np.int64)
+
+
+def set_record_run_attr(
+    cubelist: CubeList, record_run_attr: str, model_id_attr: Optional[str]
```

Comment:
The default value for `model_id_attr` in `ModalWeatherCode` is `None`, which this method now says it won't accept. We either need to not call this method in this instance, or change the typing back to include `None`.

Reply:
Are you referring to the change from `Union[str, None]` to `Optional[str]`? The use of `Optional[str]` where `None` is allowed is preferred to `Union[str, None]` by mypy: https://mypy.readthedocs.io/en/stable/kinds_of_types.html#optional-types-and-the-none-type.

Reply:
Yes, I was. I didn't know that. Thanks Gavin!